pokeapp/src/Brain.js

```diff
@@ -130,15 +136,57 @@ export const Brain = () => {
     return updatedPokeInfo;
   };
 
+  const addToPokeList = (pokemon) => {
+    const pokeIndex = myPokeList.findIndex(
+      (element) => element.id === pokemon.id
+    );
+    if (pokeIndex !== -1) {
+      let tempPokeList = [...myPokeList];
+      tempPokeList[pokeIndex].quantity++;
+      setMyPokeList(tempPokeList);
+    } else {
+      let tempPokemon = pokemon;
+      tempPokemon.quantity++;
+      setMyPokeList([...myPokeList, tempPokemon]);
+    }
+  };
+  //Not wokring on the SideBar
+  const removeFromPokeList = (pokemon, e) => {
+    //Finds the index of pokemon in the array that matches the id of the pokemon being added to the array
+    const pokeIndex = myPokeList.findIndex(
+      (element) => element.id === pokemon.id
+    );
+    let tempPokeList = [...myPokeList];
+
+    //If the pokemon is already in the array
+    if (pokeIndex !== -1) {
+      if (tempPokeList[pokeIndex].quantity > 1) {
+        tempPokeList[pokeIndex].quantity--;
+      } else if (tempPokeList[pokeIndex].quantity === 1) {
+        tempPokeList[pokeIndex].quantity--;
+        tempPokeList = tempPokeList.filter(
+          (pokemon) => pokemon.id !== parseInt(e.target.value)
+        );
+        setMyPokeList(tempPokeList);
+      }
+    }
+  };
+
+  const subtractPokeDollars = (pokemon) => {
+    setPokeDollars((currentAmount) => currentAmount - pokemon.value);
+  };
+
+  const addPokeDollars = (pokemon) => {
+    setPokeDollars((currentAmount) => currentAmount + pokemon.value);
+  };
```

Comment:
This is also global state.  We should store it in Redux rather than having one component manage the whole app.